ql/src/java/org/apache/hadoop/hive/ql/optimizer/GenMapRedUtils.java

```diff
@@ -1895,7 +1901,20 @@ public static boolean isSkewedStoredAsDirs(FileSinkDesc fsInputDesc) {
       }
 
       if ((srcDir != null) && srcDir.equals(fsopFinalDir)) {
-        return mvTsk;
+        if (isDirectInsert || isMmFsop) {
+          if (moveTaskId != null && fsoMoveTaskId != null && moveTaskId.equals(fsoMoveTaskId)) {
+            // If the ACID direct insert is on, the MoveTasks cannot be identified by the srcDir as
+            // in this case the srcDir is always the root directory of the table.
+            // We need to consider the ACID write type to identify the MoveTasks.
+            return mvTsk;
+          }
+          if ((moveTaskId == null || fsoMoveTaskId == null) && moveTaskWriteType != null
```

Comment:
Double check if this is needed

Reply:
There was a test which was failing if this was not there, but since then I think I fixed the moveTaskId generation, so cannot be null. It think this is not needed. I will remove it and let's see what the tests say.